engine/runtime/src/main/scala/org/enso/compiler/data/BindingsMap.scala

```diff
@@ -727,6 +742,14 @@ object BindingsMap {
     def toConcrete(moduleMap: ModuleMap): Option[ResolvedImport] = {
       module.toConcrete(moduleMap).map(x => this.copy(module = x))
     }
+
+    /** Determines if this resolved import statement was generated by the compiler.
+      *
+      * @return true, if generated by the compiler, false otherwise
+      */
+    def isVirtual(): Boolean = {
+      importDef.isVirtual && exports.map(_.isVirtual).getOrElse(false)
```

Comment:
It is slightly unclear to me why this is an `&&` and not an `||`? In which case can it happen that the `importDef` would be virtual but exports not?

Reply:
Both should always be virtual because both are added by the compiler at the same time.

Reply:
That's what I expected and I think that's why it got me confused why we have `&&`.
Maybe we should do 
```suggestion
      if (importDef.isVirtual) {
         assert exports.map(_.isVirtual).contains(true)
         true
      } else false
```

to express that it is an 'iff' relation?

Or ignore the assert but only check the import if we are sure export is going to be the same?
```suggestion
      importDef.isVirtual
```

The current code checks for the exports and if they were non-virtual it will return false even for a virtual import. As this situation should never happen in practice, IMO the secondary check for exports is redundant. If we want to keep it, I'd suggest to at least report (via assertion or internal compiler error) that an illegal state has been encountered.

Reply:
Fair enough, I will remove the redundant part.